pkg/vault/contracts/VaultFactory.sol

```diff
@@ -64,28 +74,67 @@ contract VaultFactory is Authentication {
      * @param targetAddress Expected Vault address. The function will revert if the given salt does not deploy the
      * Vault to the target address.
      */
-    function create(bytes32 salt, address targetAddress) external authenticate {
-        if (isDisabled) {
-            revert VaultAlreadyCreated();
+    function create(
+        bytes32 salt,
+        address targetAddress,
+        bytes memory vaultCreationCode,
+        bytes memory vaultAdminCreationCode,
+        bytes memory vaultExtensionCreationCode
```

Comment:
Perhaps it's worth using `calldata` here?

Reply:
Should save a little, and be insurance against it changing (though it clearly doesn't get changed)